pkg/agent/controller/networkpolicy/reconciler_test.go

```diff
@@ -703,3 +704,186 @@ func TestReconcilerUpdate(t *testing.T) {
+				},
+				&v1beta1.GroupMemberPod{
+					IP:    v1beta1.IPAddress(net.ParseIP("1.1.1.2")),
+					Ports: []v1beta1.NamedPort{{Port: 80, Name: "http", Protocol: protocolTCP}},
+				},
+				&v1beta1.GroupMemberPod{
+					IP:    v1beta1.IPAddress(net.ParseIP("1.1.1.3")),
+					Ports: []v1beta1.NamedPort{{Port: 8080, Name: "http", Protocol: protocolTCP}},
+				},
+				&v1beta1.GroupMemberPod{
+					IP:    v1beta1.IPAddress(net.ParseIP("1.1.1.4")),
+					Ports: []v1beta1.NamedPort{{Port: 443, Name: "https", Protocol: protocolTCP}},
+				},
+				&v1beta1.GroupMemberPod{
+					IP: v1beta1.IPAddress(net.ParseIP("1.1.1.5")),
+				},
+				&v1beta1.GroupMemberPod{
+					IP:    v1beta1.IPAddress(net.ParseIP("1.1.1.6")),
+					Ports: []v1beta1.NamedPort{{Port: 443, Name: "foo", Protocol: protocolTCP}},
+				},
+			),
+			want: map[servicesHash]v1beta1.GroupMemberPodSet{
+				hashServices([]v1beta1.Service{serviceTCP80, serviceHTTPS}): v1beta1.NewGroupMemberPodSet(
+					&v1beta1.GroupMemberPod{
+						IP:    v1beta1.IPAddress(net.ParseIP("1.1.1.1")),
+						Ports: []v1beta1.NamedPort{{Port: 80, Name: "http", Protocol: protocolTCP}},
+					},
+					&v1beta1.GroupMemberPod{
+						IP:    v1beta1.IPAddress(net.ParseIP("1.1.1.2")),
+						Ports: []v1beta1.NamedPort{{Port: 80, Name: "http", Protocol: protocolTCP}},
+					},
+				),
+				hashServices([]v1beta1.Service{serviceTCP8080, serviceHTTPS}): v1beta1.NewGroupMemberPodSet(
+					&v1beta1.GroupMemberPod{
+						IP:    v1beta1.IPAddress(net.ParseIP("1.1.1.3")),
+						Ports: []v1beta1.NamedPort{{Port: 8080, Name: "http", Protocol: protocolTCP}},
+					},
+				),
+				hashServices([]v1beta1.Service{serviceHTTP, serviceTCP443}): v1beta1.NewGroupMemberPodSet(
+					&v1beta1.GroupMemberPod{
+						IP:    v1beta1.IPAddress(net.ParseIP("1.1.1.4")),
+						Ports: []v1beta1.NamedPort{{Port: 443, Name: "https", Protocol: protocolTCP}},
+					},
+				),
+				hashServices([]v1beta1.Service{serviceHTTP, serviceHTTPS}): v1beta1.NewGroupMemberPodSet(
+					&v1beta1.GroupMemberPod{
+						IP: v1beta1.IPAddress(net.ParseIP("1.1.1.5")),
+					},
+					&v1beta1.GroupMemberPod{
+						IP:    v1beta1.IPAddress(net.ParseIP("1.1.1.6")),
+						Ports: []v1beta1.NamedPort{{Port: 443, Name: "foo", Protocol: protocolTCP}},
+					},
+				),
+			},
+			want1: map[servicesHash][]v1beta1.Service{
+				hashServices([]v1beta1.Service{serviceTCP80, serviceHTTPS}):   {serviceTCP80, serviceHTTPS},
+				hashServices([]v1beta1.Service{serviceTCP8080, serviceHTTPS}): {serviceTCP8080, serviceHTTPS},
+				hashServices([]v1beta1.Service{serviceHTTP, serviceTCP443}):   {serviceHTTP, serviceTCP443},
+				hashServices([]v1beta1.Service{serviceHTTP, serviceHTTPS}):    {serviceHTTP, serviceHTTPS},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, got1 := groupPodsByServices(tt.services, tt.pods)
+			assert.Equal(t, tt.want, got)
+			assert.Equal(t, tt.want1, got1)
```

Comment:
got/got1, want/want1 are perhaps not the best names?

Reply:
done